src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration.Tests/MockHttpClientTransport.cs

```diff
@@ -139,14 +138,18 @@ protected override void WriteResponseCore(HttpResponseMessage response)
             var bathItems = new MockBatch();
             int itemIndex = batch.index;
             int count = batch.count;
+
+            StringBuilder responseContent = new StringBuilder();
```

Comment:
Why not use JsonWriter to write JSON? We want this library to be an example for how to write client libraries. StringBuilder works ok for this simple JSON, but in probably any other client library, writing JSON using StringBuilder would be very cumbersome and bug prone.

Reply:
There is no reason to replace Newtonsoft.Json with StringBuilder in tests. If JsonWriter is too hard to use for testing might want to stick with Newtonsoft.Json as test-only dependency.

Reply:
I have added logic to use JsonWriter in the mock tests instead of StringBuilder